### ISSUE TITLE
react/prop-types false positive “is missing in props validation” for extended component types

### ISSUE THREAD

**Issue by flying-sheep** [yannickcr/eslint-plugin-react#3159]
I’m building a component with antd-style subcomponent access. I.e. I want to be able to do `<Columns><Columns.Block/></Columns>`.

I made two attempts, both of which failed:

By extending `FC<P>`:

```ts
const Block: FC<...> = (...) => ...

export interface ColumnProps { ... }
export interface ColumnsComponent extends FC<ColumnsProps> {
  Block: typeof Block
}
const Columns: ColumnsComponent = (...) => ...
Columns.Block = Block
```

and inline:

```ts
const Block: FC<...> = (...) => ...

export interface ColumnProps { ... }
const Columns: FC<ColumnsProps> & { Block: typeof Block } = (...) => ...
Columns.Block = Block
```

**Comment by gwmccull** [comment]
I'm using Typescript and React Native with the React Native Navigation library, and I have a similar issue. I found the following workaround for their `NavigationFunctionComponent` which extends the `React.FunctionComponent` type. It adds the `componentId` to the component

```typescript
import React from "react";
import { NavigationFunctionComponent } from "react-native-navigation";
import PropTypes from "prop-types";

import AccountConfirmation from "../../components/launch/AccountConfirmation";

interface Props {
  token: string;
}

const AccountConfirmationScreen: NavigationFunctionComponent<Props> = ({
  componentId,
  token,
}) => {
  return <AccountConfirmation token={token} />;
};

function generatePropTypes() {
  return {
    componentId: PropTypes.string.isRequired,
    ...PropTypes.any,
  };
}

AccountConfirmationScreen.propTypes = {
  ...generatePropTypes(),
};
```

I'll be moving that `generatePropTypes` into a utils file to be used in a bunch of places
